3-0-java-core/3-6-4-random-field-comparator/src/test/java/com/bobocode/se/RandomFieldComparatorTest.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @Test
+    @Order(8)
+    @SneakyThrows
+    @DisplayName("'compare' should return 1 if field value of first object is greater")
+    void compareWhenFieldValueOfFirstObjectIsGrater() {
+        var fieldToCompareName = "firstName";
+        Account account1 = Accounts.generateAccount();
+        Account account2 = Accounts.generateAccount();
+        Field fieldToCompareAccount = account1.getClass().getDeclaredField(fieldToCompareName);
+        fieldToCompareAccount.setAccessible(true);
+
+        fieldToCompareAccount.set(account1, "Bob");
+        fieldToCompareAccount.set(account2, "Alice");
+
+        setFieldToCompare(fieldToCompareName, Account.class);
+        int compareResult = randomFieldComparator.compare(account1, account2);
+
+        assertEquals(1, compareResult);
+    }
+
+    @Test
+    @Order(9)
+    @SneakyThrows
+    @DisplayName("'compare' should return -1 if field value of second object is greater")
+    void compareWhenFieldValueOfSecondObjectIsGrater() {
+        var fieldToCompareName = "firstName";
+        Account account1 = Accounts.generateAccount();
+        Account account2 = Accounts.generateAccount();
+        Field fieldToCompareAccount = account1.getClass().getDeclaredField(fieldToCompareName);
+        fieldToCompareAccount.setAccessible(true);
+
+        fieldToCompareAccount.set(account1, "Alice");
+        fieldToCompareAccount.set(account2, "Bob");
+
+        setFieldToCompare(fieldToCompareName, Account.class);
+        int compareResult = randomFieldComparator.compare(account1, account2);
+
+        assertEquals(-1, compareResult);
+    }
+
+    @Test
+    @Order(10)
+    @SneakyThrows
+    @DisplayName("'compare' should return -1 if field value of second object is greater")
+    void compareWhenFieldValuesOfObjectsAreEqual() {
+        var fieldToCompareName = "firstName";
+        Account account1 = Accounts.generateAccount();
+        Account account2 = Accounts.generateAccount();
+        Field fieldToCompareAccount = account1.getClass().getDeclaredField(fieldToCompareName);
+        fieldToCompareAccount.setAccessible(true);
+
+        fieldToCompareAccount.set(account1, "Carol");
+        fieldToCompareAccount.set(account2, "Carol");
+
+        setFieldToCompare(fieldToCompareName, Account.class);
+        int compareResult = randomFieldComparator.compare(account1, account2);
+
+        assertEquals(0, compareResult);
+    }
+
+    @Test
+    @Order(11)
+    @SneakyThrows
+    @DisplayName("'getComparingFieldName' should return the name of randomly-chosen field to be compared")
+    void getComparingFieldName() {
+        var fieldToCompareName = "lastName";
+        setFieldToCompare(fieldToCompareName, Account.class);
+
+        assertEquals(fieldToCompareName, randomFieldComparator.getComparingFieldName());
+    }
+
+    @Test
+    @Order(12)
+    @SneakyThrows
+    @DisplayName("'toString' should return the string - \"Random field comparator of class '%s' is comparing '%s'\"")
```

Comment:
I don't think it's a good idea to put the whole string template into the test name.
Let's just say `Method toString is properly overridden`